File: back-end/internal/entity/attendance.go
package entity

import (
	"time"
)

type Attendance struct {
	ID             		int       `db:"id" json:"id"`
	EmployeeID     		string     `db:"employee_id" json:"employee_id"`
	Date           		time.Time `db:"date" json:"date"`
	CheckInTime    		string	  `db:"check_in_time" json:"check_in_time"`
	IsLate         		bool      `db:"is_late" json:"is_late"`
	CheckOutTime   		string	  `db:"check_out_time" json:"check_out_time"`
}

type AttendanceRequest struct {
    ID              int       `db:"id"`               
    EmployeeID      string    `db:"employee_id"`
    RequestDate     time.Time `db:"request_date"`
    CheckInTime     string    `db:"check_in_time"`
    CheckOutTime    string    `db:"check_out_time"`
    Reason          string    `db:"reason"`
    Status          string    `db:"status"`
    RejectionReason string    `db:"rejection_reason"` 
    CreatedAt       time.Time `db:"created_at"`
    UpdatedAt       time.Time `db:"updated_at"`
}

File: front-end/src/app/types/Attendance.ts
export interface Attendance {
  id?: number;
  employee_id: string;
  date: string;
  check_in_time: string | null;
  is_late: boolean;
  check_out_time: string | null;
}
  
export interface AttendanceRequest {
  id: number;
  employee_id: string;
  request_date: string;
  check_in_time: string;
  check_out_time: string;
  reason: string;
  status: string;
  rejection_reason?: string;
  created_at: string;
  updated_at: string;
}
  